harness/determined/tensorboard/base.py

```diff
@@ -83,6 +109,21 @@ def delete(self) -> None:
         """
         pass
 
+    def start_async_upload_thread(self) -> None:
+        if self.upload_thread is not None and not self.upload_thread.is_alive():
```

Comment:
optional: especially for internal objects, I like objects with clear semantics.  So I would recommend not supporting duplicate start() calls.

(though usually I allow duplicate close() calls because that's just so darn useful.